### ISSUE TITLE
[Bug]: [Nightly] Error checking on diskann search_list is broken

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#28745]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 8a9ab69
- Deployment mode(standalone or cluster):standalone and cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.2.post1.dev5
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Error checking on diskann search_list is broken.

limit=1, search_list=-1 / 0 / 201 , pass
limit=20, search_list=19 / 201 , pass
limit=6553, search_list=6550 / 65536 , pass

### Expected Behavior

pass

### Milvus Log

1. link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/571/pipeline/205
2. log: [artifacts-milvus-distributed-kafka-nightly-571-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/571/artifact/artifacts-milvus-distributed-kafka-nightly-571-pymilvus-e2e-logs.tar.gz)
3. failed cases:
```
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_A[32-False-201-1] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_A[32-True-201-1] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_A[128-False-201-1] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_A[128-True-201-1] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_B[32-False-201-20] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_B[32-True-201-20] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_B[128-False-201-20] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_B[128-True-201-20] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_C[32-False-65536-6553] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_C[32-True-65536-6553] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_C[128-False-65536-6553] - AssertionError
[2023-11-26T20:10:51.186Z] FAILED testcases/test_search.py::TestSearchDiskann::test_search_invalid_params_with_diskann_C[128-True-65536-6553] - AssertionError
```

**Comment by NicoYuan1986** [comment]
> Disknn now support search with serach_list_num in [topk, 2147483647]. So some e2e case outdated.

Got it. I will change cases later.

**Comment by NicoYuan1986** [comment]
fixed by #28949
